LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/OTAAHelper/OTAAKeysGenerator.cs

```diff
@@ -36,7 +37,10 @@ public static string CalculateKey(byte[] type, byte[] appnonce, byte[] netid, by
 #pragma warning restore CA5358 // Review cipher mode usage with cryptography experts
             aes.Padding = PaddingMode.None;
 
-            var pt = type.Concat(appnonce).Concat(netid).Concat(devnonce).Concat(new byte[7]).ToArray();
+            var devNonceBytes = new byte[DevNonce.Size];
+            _ = devnonce.Write(devNonceBytes);
+
+            var pt = type.Concat(appnonce).Concat(netid).Concat(devNonceBytes).Concat(new byte[7]).ToArray();
```

Comment:
Again, we can do like suggested for `LoRaPayload.CalculateKey`.

Reply:
Done in c6e00385d1a8b7384e1e813a8c8d64b343160f05